Predicting Newsletter Subscription from player demographics and behaviour

In [18]:
library(tidyverse)

In [22]:
library(dplyr) # Assuming you are using 'dplyr' with the pipe
# Calculate the mean age
players<-read_csv("data/players.csv")
sessions<-read_csv("data/sessions.csv")
mean_stats <- players |>
  summarise(
    mean_age = mean(as.numeric(Age), na.rm = TRUE),
    mean_played_hrs = mean(as.numeric(played_hours), na.rm = TRUE)
  )

mean_stats

Rows: 196 Columns: 7
── Column specification ────────────────────────────────────────────────────────
Delimiter: ","
chr (4): experience, hashedEmail, name, gender
dbl (2): played_hours, Age
lgl (1): subscribe

ℹ Use `spec()` to retrieve the full column specification for this data.
ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.
Rows: 1535 Columns: 5
── Column specification ────────────────────────────────────────────────────────
Delimiter: ","
chr (3): hashedEmail, start_time, end_time
dbl (2): original_start_time, original_end_time

ℹ Use `spec()` to retrieve the full column specification for this data.
ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.


mean_age mean_played_hrs
1 21.13918 5.845918

1) Dataset

Players - 196 observations, 7 variables
Sessions - 1555 observations, 5 variables

Players Summary 
- experience (chr): experience of a player ranging from Beginner, Regular, Amateur, Veteran and Pro
- subscribe (lgl): whether player is subscribed to newsletter or not
- hashedemail (chr): hashed email of player
- played_hours (dbl): hours played by player
- name (chr): name of player
- gender (chr): gender of player
- Age (dbl): age of player
- mean of Age - 21.14
- mean of Hours Played - 5.85

Sessions Summary
- hashedemail (chr) - same as in Players dataframe
- start_time (chr) - start time of playing
- end_time (chr) - end time of playing
- original_start_time (dbl) - same as above in number format
- original_end_time (dbl) - same as above in number format

Issues I see:
- Sessions dataset has columns that are confusing such as start time and original start time
- Name is visible which could be a confidentiality issue
- Outliers present
- Duplicates


2)
Broad question: What player characteristics and behaviours predict subscribing to the game-related newsletter?

Specific question: Can the hours played by a player and age of a player predict whether a player subscribes to the newsletter?